common/apiclient.py

```diff
@@ -56,9 +56,9 @@ def list_objects(self, path):
 
         return workspace_path_obj
 
-    def execute_notebook(self, notebook_path, cluster_id,
-                         notebook_params=None, timeout=120,
-                         pull_wait_time=DEFAULT_POLL_WAIT_TIME):
+    def execute_notebook(self, notebook_path, cluster_id, timeout=120,
+                         pull_wait_time=DEFAULT_POLL_WAIT_TIME,
+                         notebook_params=None):
```

Comment:
I think we are missing a test here. The test should assert that when `notebook_params` are set, the underlying client call (`dbclient.jobs.submit_run`) receives the parameters in the correct format/structure.

Reply:
Hi @giventocode, possible misunderstanding, but I thought the `test__execute_notebook__non_dict_params__valueerror` test (line 80 in test_apiclient.py) would cover a non-dictionary passed error. There is also a check on line 60 of apiclient.py that ensures an error is thrown if the notebook params is not passed in as a dictionary that predates this PR. Were these incorrect assumptions to make?

Reply:
The test validates that what is received is a dictonary, but what I am wondering is at a highlevel. For instance, is it possible to receive empty values or keys?

Reply:
Ah, understood! Will add these checks and validate. Thanks @giventocode!